### PR TITLE
Prevent hiding underlying exception when ConfTestImportFailure is raised

### PR DESCRIPTION
**Pull request by DahlitzFlorian**
@asottile the PR I mentioned in #7150. Let me know if I missed something. Thanks for your patience and your help on this, I really appreciate it!

- [X] Allow maintainers to push and squash when merging my commits. Please uncheck this if you prefer to squash the commits yourself.
- [X] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`
- [X] Add yourself to `AUTHORS` in alphabetical order (was done in PR #7240 and approved, but not yet merged)

### CODE REVIEW

**Comment by symonk** [comment]
Hey @DahlitzFlorian github is having some issue(s) right now which explains the fetch failures on CI :)

File: src/_pytest/debugging.py [hunk 1]
```diff
@@ -338,6 +339,10 @@ def _postmortem_traceback(excinfo):
         # A doctest.UnexpectedException is not useful for post_mortem.
         # Use the underlying exception instead:
         return excinfo.value.exc_info[2]
+    elif isinstance(excinfo.value, ConftestImportFailure):
+        # A config.ConftestImportFailure is not useful for post_mortem.
+        # Use the underlying exception instead:
+        return excinfo.value.excinfo[2]
```
Comment (asottile) [inline src/_pytest/debugging.py:345]: nice!  that was easy -- is there a way to demo this in a test?

Comment (DahlitzFlorian) [reply src/_pytest/debugging.py:345]: Thanks for the fast feedback! Oh yeah, I will try to write a test for this today.

Comment (nicoddemus) [reply src/_pytest/debugging.py:345]: Please take a look at `test_debugging.py`, there are tests there which interact with pdb directly. 👍

**Comment by DahlitzFlorian** [comment]
Finally, I got a working test, which actually does test the change I submitted - never mind writing the test would be harder than the actual fix.

**Pull request merged**